plugins/importexport/doaj/classes/DOAJExportDom.inc.php

```diff
@@ -20,22 +20,35 @@ class DOAJExportDom {
 	 * Generate the export DOM tree for a given journal.
 	 * @param $doc object DOM object
 	 * @param $journal object Journal to export
+	 * @param $selectedObjects array
 	 */
-	function &generateJournalDom(&$doc, &$journal) {
-		$issueDao =& DAORegistry::getDAO('IssueDAO');
-		$sectionDao =& DAORegistry::getDAO('SectionDAO');
-		$pubArticleDao =& DAORegistry::getDAO('PublishedArticleDAO');
+	function generateJournalDom($doc, $journal, $selectedObjects) {
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		$sectionDao = DAORegistry::getDAO('SectionDAO');
+		$pubArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 
 		// Records node contains all articles, each called a record
-		$records =& XMLCustomWriter::createElement($doc, 'records');
+		$records = XMLCustomWriter::createElement($doc, 'records');
 
-		$pubArticles =& $pubArticleDao->getPublishedArticlesByJournalId($journal->getId());
-		while ($pubArticle =& $pubArticles->next()) {
-			$issue =& $issueDao->getIssueById($pubArticle->getIssueId());
+		$pubArticles = $pubArticleDao->getPublishedArticlesByJournalId($journal->getId());
+		while ($pubArticle = $pubArticles->next()) {
+
+			// check for selected issues:
+			$issueId = $pubArticle->getIssueId();
+			$selectedIssues = $selectedObjects[DOAJ_EXPORT_ISSUES];
```

Comment:
This will cause an uninitialized index warning, I think, as that index won't be set in all cases. (Also elsewhere.)